eventsources/eventing.go

```diff
@@ -422,6 +445,23 @@ func (e *EventSourceAdaptor) run(ctx context.Context, servers map[apicommon.Even
 						if err != nil {
 							return err
 						}
+
+						if filter, ok := filters[getFilterKey(s.GetEventSourceType(), s.GetEventName())]; ok {
+							dataMap := make(map[string]interface{})
+							err := json.Unmarshal(data, &dataMap)
+							if err != nil {
+								return err
+							}
+							proceed, err := shouldDispatch(filter.Expression, dataMap)
+							if err != nil {
+								return err
+							}
+							if !proceed {
+								logger.Error("failed to publish event, filter condition not met")
```

Comment:
Don't log the error here, it's not an error, it's just because it does not meet the filtering condition. A debug log should be good.